nautobot/docs/plugins/development.md

```diff
@@ -574,6 +575,134 @@ A `PluginMenuButton` has the following attributes:
 
+## Navigation Menu Tabs
+
+Plugins can modify the existing navigation bar layout by using `nav_tabs` inside of `navigation.py`. Using the key and weight system, a developer can modify existing tabs, groups, items and buttons.
+
+### Adding a new tab
+
+The code below shows how to add a new tab to the navbar. A tab is defined by a `NavMenuTab` object. Similarly a group is defined using `NavMenuGroup`. Both of these objects are used as containers for actual items.
+
+The position in the navigation menu is defined by the weight. The lower the weight the closer to the same the object will be. All core objects have weights in multiples of 100., meaning there is plenty of space around the objects for plugins to customize.
+
+Below you can see `Dummy Tab` has a weight value of `150`. This means the tab will appear between `Organization` and `Devices`.
+
+``` python
+from nautobot.core.apps import NavMenuAddButton, NavMenuGroup, NavMenuItem, NavMenuImportButton, NavMenuTab
+
+menu_tabs = (
+    NavMenuTab(
+        name="Dummy Tab",
+        weight=150,
+        groups=(
+            NavMenuGroup(
+                name="Dummy Group 1",
+                weight=100,
+                items=(
+                    NavMenuItem(
+                        link="plugins:dummy_plugin:dummymodel_list",
+                        link_text="Dummy Model",
+                        permissions=[
+                            "dummy_plugin.view_dummymodel"
+                        ],
+                        buttons=(
+                            NavMenuAddButton(
+                                link="plugins:dummy_plugin:dummymodel_add",
+                                permissions=[
+                                    "dummy_plugin.add_dummymodel",
+                                ],
+                            ),
+                            NavMenuImportButton(
+                                link="plugins:dummy_plugin:dummymodel_import",
+                                permissions=[
+                                    "dummy_plugin.add_dummymodel"
+                                ],
+                            ),
+                        ),
+                    ),
+                ),
+            ),
+        ),
+    ),
+)
+```
+
+### Modifying existing tab
+
+By defining an object with the same identifier, a developer can modify existing objects. The example below shows modifying an existing tab to have a new group.
+
+A tab object is being created with the same identifer as an existing object using the `name` attribute. Then a group is being created with a weight of `150`, which means it will appear between the already defined `Circuits` and `Provider` groups.
+
+This pattern works for modifying all objects in the tree. New items can be added to groups and new buttons can be added to items.
+
+``` python
+menu_tabs = (
+    NavMenuTab(
+        name="Circuits",
+        groups=(
```

Comment:
Thinking about how things were implemented in #485, I think this will actually result with errors being logged from `check_or_create_entry`, as presently implemented, because the `weight` of this `NavMenuTab` doesn't match the weight previously defined for `Circuits`.

 I do think we want to permit this approach (you shouldn't need to know the weight of an existing tab/group/item in order to augment it) but I think it will require some code changes to do so.

Reply:
This feature works. Tested in the dummy plugin

``` python
    NavMenuTab(
        name="Circuits",
        groups=(
            NavMenuGroup(
                name="Dummy Circuit Group",
                weight=150,
                items=(
                    NavMenuItem(
                        link="plugins:dummy_plugin:dummymodel_list",
                        link_text="Dummy Model",
                        permissions=["dummy_plugin.view_dummymodel"],
                        buttons=(
                            NavMenuAddButton(
                                link="plugins:dummy_plugin:dummymodel_add",
                                permissions=[
                                    "dummy_plugin.add_dummymodel",
                                ],
                            ),
                            NavMenuImportButton(
                                link="plugins:dummy_plugin:dummymodel_import",
                                permissions=["dummy_plugin.add_dummymodel"],
                            ),
                        ),
                    ),
                ),
            ),
        ),
    ),
```

Reply:
And you confirmed that no `error` or `warning` messages are logged to the nautobot logs in this case? I'd really expect that we'd be hitting https://github.com/nautobot/nautobot/blob/develop-1.1.0/nautobot/core/apps/__init__.py#L43 in this scenario.

Reply:
No errors or warnings

Reply:
Ok, can you help me understand why this is the case? Here's my understanding of why an error *should* be logged:

-  `NavMenuTab(name="Circuits", groups=...)` will default to a `weight` of `1000`.
- This is different than the prior declaration of the Circuits tab in `circuits/navigation.py` (which explicitly sets `weight=500`. 
- When we process the plugin `menu_items`, we call `create_or_check_entry()` on the `NavMenuTab`, which includes `("weight", 1000)` amongst its `fixed_fields`.
- This doesn't match the previously registered weight of 500, so a `logger.error()` should be triggered.

Reply:
Ah i see whats happening. For some reason my log messages are not coming through my terminal anymore. There is an error happening, so when the plugin defines `circuits` the weight value defaults to `1000`. This logs an error saying the weight was originally set to `150` but for some reason this wasnt being showed in my `stdout`